hashicorp/terraform-provider-tfe
# feat(workspace): add auto destroy activity duration

## Description

Adds `auto_destroy_activity_duration` to the tfe_workspace resource.

~~Requires #1378~~

_Remember to:_


## Testing plan

1.  create a workspace resource with `auto_destroy_activity_duration`
    - example from test:
     ```
     resource "tfe_workspace" "foobar" {
       name                           = "workspace-test"
       organization                   = tfe_organization.foobar.id
       auto_apply                     = true
       file_triggers_enabled           = false
       auto_destroy_activity_duration = "1d"
     }
     ```
1. apply configuration
1. confirm that there are no pending changes on subsequent applies

## External links

_Include any links here that might be helpful for people reviewing your PR. If there are none, feel free to delete this section._

- [API documentation](https://developer.hashicorp.com/terraform/cloud-docs/api-docs/workspaces#request-body)
- [Related PR](https://github.com/hashicorp/terraform-provider-tfe/pull/1354)

## Output from acceptance tests

_Please run applicable acceptance tests locally and include the output here. See [testing.md](https://github.com/hashicorp/terraform-provider-tfe/blob/main/docs/testing.md) to learn how to run acceptance tests._

_If you are an external contributor, your contribution(s) will first be reviewed before running them against the project's CI pipeline._

```
$ TFE_HOSTNAME=<hostname> TFE_TOKEN=<token> TESTARGS='-run "TestAccTFEWorkspace(DataSource)?_(create|update|read)(With)?AutoDestroy(At|Duration)"' make testacc
TF_ACC=1 TF_LOG_SDK_PROTO=OFF go test $(go list ./... |grep -v 'vendor') -v -run "TestAccTFEWorkspace(DataSource)?_(create|update|read)(With)?AutoDestroy(At|Duration)" -timeout 15m
?   	github.com/hashicorp/terraform-provider-tfe	[no test files]
testing: warning: no tests to run
PASS
ok  	github.com/hashicorp/terraform-provider-tfe/internal/client	0.205s [no tests to run]
testing: warning: no tests to run
PASS
ok  	github.com/hashicorp/terraform-provider-tfe/internal/logging	0.269s [no tests to run]
?   	github.com/hashicorp/terraform-provider-tfe/internal/provider/validators	[no test files]
?   	github.com/hashicorp/terraform-provider-tfe/version	[no test files]
=== RUN   TestAccTFEWorkspaceDataSource_readAutoDestroyAt
--- PASS: TestAccTFEWorkspaceDataSource_readAutoDestroyAt (9.94s)
=== RUN   TestAccTFEWorkspaceDataSource_readAutoDestroyDuration
--- PASS: TestAccTFEWorkspaceDataSource_readAutoDestroyDuration (9.72s)
=== RUN   TestAccTFEWorkspace_createWithAutoDestroyAt
--- PASS: TestAccTFEWorkspace_createWithAutoDestroyAt (6.47s)
=== RUN   TestAccTFEWorkspace_updateWithAutoDestroyAt
--- PASS: TestAccTFEWorkspace_updateWithAutoDestroyAt (10.93s)
=== RUN   TestAccTFEWorkspace_createWithAutoDestroyDuration
--- PASS: TestAccTFEWorkspace_createWithAutoDestroyDuration (5.22s)
=== RUN   TestAccTFEWorkspace_updateWithAutoDestroyDuration
--- PASS: TestAccTFEWorkspace_updateWithAutoDestroyDuration (13.36s)
PASS
ok  	github.com/hashicorp/terraform-provider-tfe/internal/provider	56.030s
```
